### PR TITLE
QT6 build for Apple M1

### PR DESCRIPTION
**Pull request by WebmasterTD**
This is an experimental version to convert the project to QT version 6.21.
I was able to build for Apple M1 with these modifications.
It has some not finished conversions from QT5 to QT6, marked them with //TODO
I intend this as a guide/overview what it takes to convert the project. This is not polished. Still has some bugs.

### CODE REVIEW

**Comment by limpkin** [comment]
thanks a lot :)

File: src/CredentialsManagement.cpp [hunk 1]
```diff
@@ -903,8 +903,8 @@ void CredentialsManagement::updateLoginDescription(LoginItem *pLoginItem)
             ui->credDisplayLoginInput->setText(pLoginItem->name());
             ui->credDisplayPasswordInput->setText(pLoginItem->password());
             ui->credDisplayDescriptionInput->setText(pLoginItem->description());
-            ui->credDisplayCreationDateInput->setText(pLoginItem->updatedDate().toString(Qt::DefaultLocaleShortDate));
-            ui->credDisplayModificationDateInput->setText(pLoginItem->accessedDate().toString(Qt::DefaultLocaleShortDate));
+            ui->credDisplayCreationDateInput->setText(pLoginItem->updatedDate().toString());
+            ui->credDisplayModificationDateInput->setText(pLoginItem->accessedDate().toString());
```
Comment (raoulh) [inline src/CredentialsManagement.cpp:907]: `toString(QLocale::system().dateFormat(QLocale::ShortFormat))`

File: src/MPDevice_emul.cpp [hunk 2]
```diff
@@ -19,7 +19,7 @@
 #include "MPDevice_emul.h"
 #include "MooltipassCmds.h"
 
-#define CLEAN_MEMORY_QBYTEARRAY(d) do {for (int i = 0; i < d.size(); i++){d[i] = qrand() % 256;}} while(0);
+#define CLEAN_MEMORY_QBYTEARRAY(d) do {for (int i = 0; i < d.size(); i++){d[i] = rand() % 256;}} while(0);
```
Comment (raoulh) [inline src/MPDevice_emul.cpp:22]: https://doc.qt.io/qt-5/qrandomgenerator.html

File: src/MPDevice_emul.cpp [hunk 3]
```diff
@@ -209,7 +209,7 @@ void MPDevice_emul::platformWrite(const QByteArray &data)
          d[0] = 32;
          d[1] = commandData;
          for (int i = 0; i < 32; i++)
-             d[2 + i] = qrand() % 255;
+             d[2 + i] = rand() % 255;
```
Comment (raoulh) [inline src/MPDevice_emul.cpp:212]: https://doc.qt.io/qt-5/qrandomgenerator.html

File: src/SimpleCrypt/SimpleCrypt.cpp [hunk 4]
```diff
@@ -31,14 +31,15 @@ SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
...
+#include <QIODevice>
 
 SimpleCrypt::SimpleCrypt():
     m_key(0),
     m_compressionMode(CompressionAuto),
     m_protectionMode(ProtectionChecksum),
     m_lastError(ErrorNoError)
 {
-    qsrand(uint(QDateTime::currentMSecsSinceEpoch() & 0xFFFF));
+    srand(uint(QDateTime::currentMSecsSinceEpoch() & 0xFFFF));
```
Comment (raoulh) [inline src/SimpleCrypt/SimpleCrypt.cpp:42]: https://doc.qt.io/qt-5/qrandomgenerator.html

File: src/SimpleCrypt/SimpleCrypt.cpp [hunk 5]
```diff
@@ -47,7 +48,7 @@ SimpleCrypt::SimpleCrypt(quint64 key):
...
```
Comment (raoulh) [inline src/SimpleCrypt/SimpleCrypt.cpp:51]: https://doc.qt.io/qt-5/qrandomgenerator.html

File: src/SimpleCrypt/SimpleCrypt.cpp [hunk 6]
```diff
@@ -113,7 +114,7 @@ QByteArray SimpleCrypt::encryptToByteArray(QByteArray plaintext)
     }
 
     //prepend a random char to the string
-    char randomChar = char(qrand() & 0xFF);
+    char randomChar = char(rand() & 0xFF);
```
Comment (raoulh) [inline src/SimpleCrypt/SimpleCrypt.cpp:117]: https://doc.qt.io/qt-5/qrandomgenerator.html

File: src/qtcsv/include/qtcsv/reader.h [hunk 7]
```diff
@@ -2,12 +2,13 @@
...
-#include <QTextCodec>
-
+//#include <QtCore5Compat/QTextCodec>
+#include <QStringConverter>
+#include <QStringView>
 #include "qtcsv/qtcsv_global.h"
 
 class QIODevice;
-class QStringList;
+class QStringList_o;
```
Comment (raoulh) [inline src/qtcsv/include/qtcsv/reader.h:11]: What is this `QStringList_o` ?

File: src/qtcsv/sources/writer.cpp [hunk 8]
```diff
@@ -180,7 +181,7 @@ QString WriterPrivate::getTempFileName()
 
     for (int counter = 0; counter < std::numeric_limits<int>::max(); ++counter)
     {
-        QString name = nameTemplate.arg(QString::number(qrand()));
+        QString name = nameTemplate.arg(QString::number(rand()));
```
Comment (raoulh) [inline src/qtcsv/sources/writer.cpp:184]: https://doc.qt.io/qt-5/qrandomgenerator.html

**Comment by limpkin** [comment]
done in https://github.com/mooltipass/moolticute/pull/964

**Pull request closed without merging**